roxie/roxiemem/roxiemem.cpp

```diff
@@ -472,10 +472,7 @@ static void dumpHeapState()
     for (unsigned i = 0; i < heapBitmapSize; i++)
     {
         heap_t t = heapBitmap[i];
-        if (sizeof(heap_t) == 8)
-            s.appendf("%08x", (unsigned)t);
-        else
-            s.appendf("%016" I64F "x", (unsigned __int64)t);
+        s.appendf("%0*" I64F "x", (int)(sizeof(heap_t)*2), (unsigned __int64)t);
```

Comment:
Yes, because of t is type unsigned it would be passed with the wrong length to the printf - potentially generating random values in the top part of the number.